backend/dataall/modules/dashboards/api/dashboard_schema.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import field, dataclass
+
+from dataall.base.api import gql
+from dataall.base.api.graphql_api import api_input, api_object, Page, PageFilter
+from dataall.core.environment.db.models import Environment
+from dataall.core.organizations.db.organization_models import Organization
+from dataall.modules.dashboards import Dashboard
+from dataall.modules.dashboards.api.enums import DashboardRole
+
+
+@api_object("Dashboard")
+@dataclass
+class DashboardDto(gql.ObjectType):
+    dashboardUri: gql.ID
+    name: str
+    label: str
+    description: str
+    DashboardId: str
+    tags: [str]
+    created: str
+    updated: str
+    owner: str
+    SamlGroupName: str
+    organization: gql.Ref('Organization') = None
+    environment: gql.Ref('Environment') = None
```

Comment:
I don't understand how this is replacing the resolve_environment function

Reply:
This is one of the tricky stuff I have mentioned above
It's not a replacement for `resolve_environment`. It just says that `DashboardDto` (change the name for it to avoid collision with `models.Dashboard`) has type with the reference to environment. **= None** means that the value is optional and isn't supplied all the time (can vary from method to method)

Why so? `DashboardDto` returned from a couple of functions: `import_dashboard`, `update_dashboard`, and `get_dashboard`. Only `get_dashboard` returns all the values. 
If you take a look at the `get_dashboard` you will see that `DashboardDto` is created with all information. It has dashboard, upvotes, terms, organization, env, and role.
organization and environment use the same type as before: `gql.Ref('Organization')` and `gql.Ref('Environment')` accordingly. We are resolving environment in the `get_dashboard` function
If we remove the default value for environment: `gql.Ref('Environment')` the functions `import_dashboard` and `update_dashboard` fails since they will try to return incomplete object.


The PR just mimics the previous schema, but it can be improved further since the objects and input types can be extended. I will demonstrate it here and update the PR to reflect this:
```
@api_object("Dashboard")
@dataclass
class DashboardView(gql.ObjectType): (renamed from DashboardDto)
    dashboardUri: gql.ID
    name: str
    label: str
    description: str
    DashboardId: str
    tags: [str]
    created: str
    updated: str
    owner: str


@api_object("DashboardOverview")
@dataclass
class DashboardOverview(DashboardView):  <------ Here we are extending the schema of DashboardView obj
    organization: gql.Ref('Organization')
    environment: gql.Ref('Environment') 
    userRoleForDashboard: DashboardRole
    terms: [str] 
    upvotes: int

    def __init__(
        self,
        dashboard: Dashboard,
        organization: Organization,
        role: DashboardRole,
        environment: Environment,
        terms,
        upvotes: int
    ):
        for attr in Dashboard.__dict__:
            if not attr.startswith('_'):
                setattr(self, attr, getattr(dashboard, attr))

        self.environment = environment
        self.organization = organization
        self.role = role
        self.upvotes = upvotes
        self.terms = terms
        self.userRoleForDashboard = role

```
`DashboardView` can be returned from `import_dashboard` and `update_dashboard` while `DashboardOverview` from `get_dashboard`


If all objects and input types are migrated we can remove `gql.Ref('Organization')` and replace it with `OrganizationView` or something like that